torchmetrics/text/bert.py

```diff
@@ -0,0 +1,125 @@
+        - `nthreads` (int): number of threads
+        - `batch_size` (int): bert score processing batch size
+        - `lang` (str): language of the sentences; has to specify
+                  at least one of `model_type` or `lang`. `lang` needs to be
+                  specified when `rescale_with_baseline` is True.
+        - `return_hash` (bool): return hash code of the setting
+        - `rescale_with_baseline` (bool): rescale bertscore with pre-computed baseline
+        - `baseline_path` (str): customized baseline file
+
+    Returns:
+        - precision: Precision.
+        - recall: Recall.
+        - f1: F1 score.
+        - hashcode: Hashcode of the library.
+
+    Example:
+        >>> predictions = ["hello there", "general kenobi"]
+        >>> references = ["hello there", "general kenobi"]
+        >>> results = bert_score(predictions=predictions, references=references, lang="en")
+        >>> print([round(v, 2) for v in results["f1"]])
+        [1.0, 1.0]
+    """
+
+    def __init__(
+        self,
+        model_type: Optional[str] = None,
+        lang: str = "en",
+        num_layers: int = None,
+        verbose: bool = False,
+        idf: bool = False,
+        device: Optional[str] = None,
+        batch_size: int = 64,
+        nthreads: int = 4,
+        all_layers: bool = False,
+        rescale_with_baseline: bool = False,
+        baseline_path: Optional[str] = None,
+        compute_on_step: bool = True,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Callable = None,
+    ):
+        super().__init__(
+            compute_on_step=compute_on_step,
+            dist_sync_on_step=dist_sync_on_step,
+            process_group=process_group,
+            dist_sync_fn=dist_sync_fn,
+        )
+        self.baseline_path = baseline_path
+        self.rescale_with_baseline = rescale_with_baseline
+        self.lang = lang
+        self.all_layers = all_layers
+        self.nthreads = nthreads
+        self.batch_size = batch_size
+        self.device = device
+        self.idf = idf
+        self.verbose = verbose
+        self.num_layers = num_layers
+        self.model_type = model_type
+        self.add_state("predictions", [], dist_reduce_fx="cat")
+        self.add_state("references", [], dist_reduce_fx="cat")
+
+    def update(self, predictions: List, references: List) -> None:  # type: ignore
+        """Store predictions/references for computing BERT scores.
+
+        Args:
+            predictions: List of predicted sentences
+            references: List of refernces
+        """
+        self.predictions.append(predictions)
+        self.references.append(references)
+
+    def compute(self) -> Dict:
+        """Calculate Bertscores.
+
+        Return:
+            Dict with Bertscores.
+        """
+        return bert_score(
+            predictions=self.predictions[0],
+            references=self.references[0],
```

Comment:
it seems wrong to me that we only want to give the first element of the updated list to the `bert_score` function. You probably need some flattening function (see other comment) to go from a list of list to a single list of strings
```suggestion
            predictions=_flatten(self.predictions),
            references=_flatten(self.references),
```